src/Middleware/ResponseCompression/test/BodyWrapperStreamTests.cs

```diff
@@ -17,7 +17,7 @@ public class BodyWrapperStreamTests
     {
         [Theory]
         [InlineData(null, "Accept-Encoding")]
-        [InlineData("", "Accept-Encoding")]
+        [InlineData("", ",Accept-Encoding")]
```

Comment:
This is not desirable output. An empty element should be ignored/replaced.

Reply:
Change `StringValues` to not output empty strings?

Reply:
Or Headers.Append to ignore an empty value?

Reply:
Empty string is different from not present, as you highlighted https://github.com/aspnet/AspNetCore/pull/9359#discussion_r275417254

> Does this maintain/test the distinction between there being no host header and there being an empty host header? The spec makes a distinction.

Reply:
A present but empty header has a meaning in the case of Host. Empty values in a header list do not have a meaning and should be suppressed. (e.g. ", , ,,, , , value" is the same as "value"). We do that when writing response headers to the wire, no?

Reply:
> We do that when writing response headers to the wire, no?

Yes, though empty values can be happily added to the header; which is likely why they are skipped on output.

Will make change to `StringValues` to skip them also for `.ToString()`; if the header has a real value

Reply:
https://github.com/aspnet/Extensions/pull/1696

Reply:
Changed